tests/lang/s02_core/s99_coroutines/t06_cancel_coroutine.nim

```diff
@@ -0,0 +1,28 @@
+
+## A coroutine can be cancelled, by raising an exception and letting it
+## propagate to the edge of the coroutine. Upon cancellation, the
+## ``resume`` call that started execution changes the state of the instance
+## to "aborted" and returns the instance.
+
+proc coro(cancel: bool) {.coroutine.}
+  if cancel:
+    raise CatchableError.newException("cancel")
+
+let instance = launch coro(true)
+doAssert instance.status == csSuspended
+
+resume(instance)
+doAssert instance.status == csAborted
+
+## If a coroutine instance is cancelled, the exception that caused
+## cancellation is stored in the coroutine instance object. It can be
+## extracted by using the built-in ``unwrap``.
```

Comment:
Because presently the exception is doing double duty for exceptions + cancellation. At the moment, I think cancellation needs its own thing, but that needs syntax/some operator and still jive with exceptions, as those can be raised by calls subordinate to the coroutine.

What should be possible is assuming there is an executing coroutine (`coro`) and it encounters an error: it should be able to signal to the caller than an error occurred, the caller should be able to recover, and then it can resume execution of `coro`.